src/Magnum/Audio/Listener.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+    This file is part of Magnum.
+
+    Copyright © 2010, 2011, 2012, 2013, 2014, 2015
+              Vladimír Vondruš <mosra@centrum.cz>
+    Copyright © 2015 Jonathan Hale <squareys@googlemail.com>
+
+    Permission is hereby granted, free of charge, to any person obtaining a
+    copy of this software and associated documentation files (the "Software"),
+    to deal in the Software without restriction, including without limitation
+    the rights to use, copy, modify, merge, publish, distribute, sublicense,
+    and/or sell copies of the Software, and to permit persons to whom the
+    Software is furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included
+    in all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+#include "Magnum/Audio/Listener.h"
+#include "Magnum/Audio/Playable.h"
+#include "Magnum/Audio/PlayableGroup.h"
+
+
+namespace Magnum { namespace Audio {
+
+using namespace SceneGraph;
+
+namespace Implementation {
+    void* activeListener = nullptr;
+}
+
+namespace {
+
+/* Create a Matrix4 from a Matrix3/Matrix4 */
+Matrix4 padMatrix4(const Matrix3&);
+Matrix4 padMatrix4(const Matrix4&);
+
+Matrix4 padMatrix4(const Matrix3& m) {
+    return Matrix4{Vector4::pad(m[0]), Vector4::pad(m[1]), Vector4::pad(m[2]), Vector4{}};
+}
+
+Matrix4 padMatrix4(const Matrix4& m) {
+    return m;
+}
+
+}
+
+template<UnsignedInt dimensions> void Listener<dimensions>::clean(const MatrixTypeFor<dimensions, Float>& absoluteTransformationMatrix) {
+    if(!isActive()) {
+        /* Only clean if this Listener is active */
+        return;
+    }
+
+    Renderer::setListenerPosition(soundTransformation().transformVector(Vector3::pad(absoluteTransformationMatrix.translation())));
+
+    const Vector3 fwd = soundTransformation().transformVector(
+                padMatrix4(absoluteTransformationMatrix).rotationScaling()*Vector3{0.0, 0.0, 1.0});
+    const Vector3 up = soundTransformation().transformVector(
+                padMatrix4(absoluteTransformationMatrix).rotationScaling()*Vector3{0.0, 1.0, 0.0});
```

Comment:
I believe it should be `fwd = soundTransformation().transformVector(padMatrix4(absoluteTransformationMatrix).backward())`, probably because OpenALs default fwd is `{0.0f, 0.0f, -1.0f}`, and yes, that seems to be the same.